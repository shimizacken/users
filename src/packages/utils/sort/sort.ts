const compareFn = (a: string, b: string) => {
    const nameA = a.toUpperCase();
    const nameB = b.toUpperCase();
    if (nameA < nameB) {
        return -1;
    }
    if (nameA > nameB) {
        return 1;
    }

    return 0;
};

export const sortByText = (values: Array<string>) => values.sort(compareFn);
