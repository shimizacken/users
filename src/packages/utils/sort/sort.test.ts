import {sortByText} from './sort';

describe('sort test suite', () => {
    const values = [
        {value: 'Leanne Graham'},
        {value: 'Ervin Howell'},
        {value: 'Clementine Bauch'},
        {value: 'Patricia Lebsack'},
        {value: 'Chelsey Dietrich'},
        {value: 'Mrs. Dennis Schulist'},
    ];

    it('should return new neutron', () => {
        const result = sortByText(values.map(o => o.value));

        expect(result).toMatchSnapshot();
    });
});
