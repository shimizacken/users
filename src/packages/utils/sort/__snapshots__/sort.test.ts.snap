// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sort test suite should return new neutron 1`] = `
Array [
  Object {
    "value": "Chelsey Dietrich",
  },
  Object {
    "value": "Clementine Bauch",
  },
  Object {
    "value": "Ervin Howell",
  },
  Object {
    "value": "Leanne Graham",
  },
  Object {
    "value": "Mrs. Dennis Schulist",
  },
  Object {
    "value": "Patricia Lebsack",
  },
]
`;
